extensions/HyperVExtension/src/HyperVExtension/Models/VirtualMachineCreation/VMImageSelectionValidator.cs

```diff
@@ -0,0 +1,156 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using HyperVExtension.Common;
+using HyperVExtension.Models.VMGalleryJsonToClasses;
+using Windows.Win32;
+using static HyperVExtension.Constants;
+using static HyperVExtension.Helpers.BytesHelper;
+
+namespace HyperVExtension.Models.VirtualMachineCreation;
+
+/// <summary>
+/// Used to validate all prerequisites needed to create a VM from the VM gallery.
+/// </summary>
+public class VMImageSelectionValidator
+{
+    private readonly string _imageName;
+
+    private readonly VMGalleryDisk _imageDiskInfo;
+
+    private readonly IStringResource _stringResource;
+
+    private readonly string _defaultVirtualDiskPath;
+
+    private readonly List<Func<bool>> _validationMethods;
+
+    private readonly string _defaultVirtualDiskDriveRootPath;
+
+    private ulong _defaultVirtualDiskDriveFreeSpace;
+
+    public string ErrorMessage { get; private set; } = string.Empty;
+
+    public VMImageSelectionValidator(
+        VMGalleryImage image,
+        IStringResource stringResource,
+        string defaultVirtualDiskLocation)
+    {
+        _imageName = image.Name;
+        _imageDiskInfo = image.Disk;
+        _stringResource = stringResource;
+        _defaultVirtualDiskPath = defaultVirtualDiskLocation;
+        _defaultVirtualDiskDriveRootPath = Path.GetPathRoot(defaultVirtualDiskLocation) ?? string.Empty;
+        _validationMethods = new List<Func<bool>>()
+        {
+            ValidateDefaultVirtualDiskLocation,
+            ValidateUserHasEnoughFreeSpace,
+        };
+    }
+
+    /// <summary>
+    /// To create a VM from the VM gallery we need to download an archive file that contains the virtual
+    /// disk the OS image is stored on. The download is placed in the users temp folder and the extracted
+    /// virtual disk file is placed in the users default Hyper-V virtual disk location. Note: the drive
+    /// the default Hyper-V virtual disk is located on can be different from the drive the users temp folder
+    /// is located on.
+    /// </summary>
+    private bool ValidateUserHasEnoughFreeSpace()

```

Comment:
I understand why validation has 4 checks.  This only needs 2 though, right?
1. Is the archive location big enough?
2. Is the extracted location big enough?